pkg/apis/core/validation/validation.go

```diff
@@ -1564,7 +1564,9 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 		if len(csi.ControllerPublishSecretRef.Name) == 0 {
 			allErrs = append(allErrs, field.Required(fldPath.Child("controllerPublishSecretRef", "name"), ""))
 		} else {
-			allErrs = append(allErrs, ValidateDNS1123Label(csi.ControllerPublishSecretRef.Name, fldPath.Child("name"))...)
+			for _, msg := range ValidateSecretName(csi.ControllerPublishSecretRef.Name, false) {
```

Comment:
is this value propagated to any other API fields that also have tight validation, or combined with other values and saved in API fields in ways that would be broken by allowing longer values for the secret name here?

Reply:
afaict, only csi volume source ( which can be part of inline volumes) and pv.spec.csi (for pvc claims) are the affected areas. however I would let Jan to share his thoughts as well.